src/_guides/libraries/library-tour.md

```diff
@@ -1758,93 +1761,48 @@ Traditionally, AJAX-style apps make heavy use of HttpRequest. Use
-browser, the HttpRequest class can’t easily use resources that are
-hosted on an origin that is different from the origin of the app. If you
-need to access resources that live on a different web server, you need
-to either use a technique called JSONP or enable CORS headers on the
-remote resources.
 
 #### Getting data from the server
 
 The HttpRequest static method `getString()` is an easy way to get data
 from a web server. Use `await` with the `getString()` call
 to ensure that you have the data before continuing execution.
 
-<!-- library-tour/get-data-from-server/web/main.dart -->
+<?code-excerpt "test/library_tour/html_test.dart (getString)" replace="/await.*;/[!$&!]/g"?>
 {% prettify dart %}
-import 'dart:html';
-import 'dart:async';
-
-// A JSON-formatted file next to this page.
-var uri = 'data.json';
-
-main() async {
-  // Read a JSON file.
-  var data = await HttpRequest.getString(uri);
-  processString(data);
-}
-
-processString(String jsonText) {
-  parseText(jsonText);
+Future main() async {
+  String pageHtml = [!await HttpRequest.getString(url);!]
+  // Do something with pageHtml...
 }
 {% endprettify %}
 
-Information about the JSON API is in the
-[dart:convert section](#dartconvert---decoding-and-encoding-json-utf-8-and-more).
-
-{% comment %}
-{PENDING: convert to async exception catcher?}
-{% endcomment %}
-
 Use try-catch to specify an error handler:
 
-<!-- library-tour/get-data-from-server/web/main.dart -->
+<?code-excerpt "lib/library_tour/html.dart (try-getString)"?>
 {% prettify dart %}
 try {
-  data = await HttpRequest.getString(jsonUri);
-  processString(data);
+  var data = await HttpRequest.getString(jsonUri);
+  // Process data...
 } catch (e) {
-  handleError(e);
-}
-// ...
-handleError(error) {
-  print('Uh oh, there was an error.');
-  print(error.toString());
+  // Handle exception...
 }
 {% endprettify %}
 
 If you need access to the HttpRequest, not just the text data it
 retrieves, you can use the `request()` static method instead of
 `getString()`. Here’s an example of reading XML data:
 
-<!-- library-tour/get-data-from-server/web/main.dart -->
+<?code-excerpt "test/library_tour/html_test.dart (request)" replace="/await.*;/[!$&!]/g"?>
 {% prettify dart %}
-import 'dart:html';
-import 'dart:async';
-
-// An XML-formatted file next to this page.
-var xmlUri = 'data.xml';
-
-main() async {
-  // Read an XML file.
-  try {
-    var data = await HttpRequest.request(xmlUri);
-    processRequest(data);
-  } catch (e) {
-    handleError(e);
-  }
-}
-
-processRequest(HttpRequest request) {
-  var xmlDoc = request.responseXml;
-  try {
-    var license = xmlDoc.querySelector('license').text;
-    print('License: $license');
-  } catch (e) {
-    print("$xmlUri doesn't have correct XML formatting.");
+Future main() async {
+  HttpRequest req = await HttpRequest.request(
+    url,
+    method: 'HEAD',
+  );
+  if (req.status == 200) {
+    // Successful URL access...
```

Comment:
Again, this example has been simplified, focusing on the `HttpRequest.request()` call. For variability, I've shown a HEAD method call.